lib/msal-browser/src/interaction_handler/RedirectHandler.ts

```diff
@@ -2,55 +2,56 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License.
  */
-import { IInteractionHandler } from "./IInteractionHandler";
+import { InteractionHandler } from "./InteractionHandler";
 import { BrowserAuthError } from "../error/BrowserAuthError";
-import { StringUtils, AuthorizationCodeModule, TemporaryCacheKeys, AuthenticationParameters, AuthError, TokenResponse, Constants } from "msal-common";
+import { StringUtils, AuthorizationCodeModule, TemporaryCacheKeys, TokenResponse } from "msal-common";
 import { AuthCallback } from "../app/PublicClientApplication";
 import { BrowserConfigurationAuthError } from "../error/BrowserConfigurationAuthError";
 import { BrowserStorage } from "../cache/BrowserStorage";
 import { BrowserConstants } from "../utils/BrowserConstants";
 
-export class RedirectHandler extends IInteractionHandler {
+export class RedirectHandler extends InteractionHandler {
 
     private authCallback: AuthCallback;
+    private navigateToLoginRequestUrl: boolean;
 
-    constructor(authCodeModule: AuthorizationCodeModule, storageImpl: BrowserStorage,  redirectCallback: AuthCallback) {
+    constructor(authCodeModule: AuthorizationCodeModule, storageImpl: BrowserStorage, redirectCallback: AuthCallback, navigateToLoginRequestUrl: boolean) {
         super(authCodeModule, storageImpl);
         if (!redirectCallback) {
             throw BrowserConfigurationAuthError.createRedirectCallbacksNotSetError();
         }
         this.authCallback = redirectCallback;
+        this.navigateToLoginRequestUrl = navigateToLoginRequestUrl;
     }
 
     /**
      * Redirects window to given URL.
      * @param urlNavigate 
      */
-    showUI(authRequest: AuthenticationParameters): void {
+    showUI(requestUrl: string): Window {
         this.browserStorage.setItem(TemporaryCacheKeys.ORIGIN_URI, window.location.href);
-        this.authModule.createLoginUrl(authRequest).then((urlNavigate) => {
-            this.browserStorage.setItem(BrowserConstants.INTERACTION_STATUS_KEY, BrowserConstants.INTERACTION_IN_PROGRESS);
-            // Navigate if valid URL
-            if (urlNavigate && !StringUtils.isEmpty(urlNavigate)) {
-                this.authModule.logger.infoPii("Navigate to:" + urlNavigate);
-                window.location.assign(urlNavigate);
-            } else {
-                this.authModule.logger.info("Navigate url is empty");
-                throw BrowserAuthError.createEmptyRedirectUriError();
-            }
-        });
+        this.browserStorage.setItem(BrowserConstants.INTERACTION_STATUS_KEY, BrowserConstants.INTERACTION_IN_PROGRESS);
+        // Navigate if valid URL
+        if (requestUrl && !StringUtils.isEmpty(requestUrl)) {
+            this.authModule.logger.infoPii("Navigate to:" + requestUrl);
+            window.location.assign(requestUrl);
```

Comment:
I think it would be good to abstract the code to redirect (any references to `window.location` to peform a redirect) inside a helper function somewhere.

Reply:
What is the value add for this? Do we have environments where window.location does not have the assign APIs?

Reply:
done.